internal/locate/region_cache.go

```diff
@@ -1723,6 +1730,59 @@ func (c *RegionCache) GetTiFlashStores() []*Store {
 	return stores
 }
 
+// GetTiFlashMPPStores returns all stores with lable <engine, tiflash_mpp>.
+func (c *RegionCache) GetTiFlashMPPStores(bo *retry.Backoffer) (res []*Store, err error) {
+	c.tiflashMPPStoreMu.RLock()
+	needReload := c.tiflashMPPStoreMu.needReload
+	stores := c.tiflashMPPStoreMu.stores
+	c.tiflashMPPStoreMu.RUnlock()
+
+	if needReload {
```

Comment:
So in a bad case, there may be many threads calling this function and turned to reload, however one reload task is enough.